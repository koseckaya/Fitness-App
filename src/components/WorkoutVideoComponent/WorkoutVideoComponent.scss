@import '../../styles/vars.scss';

.workout-video {
  display: flex;
  position: relative;
  width: 32%;
  a {
    width: 100%;
  }
  &__img {
    width: 100%;
    box-shadow: rgb(38, 57, 77) 0px 15px 20px -10px;
    &:hover {
      transition: .3s;
      transform: scale(1.005);
      background-color: rgba(102, 67, 1, 0.716);
    }
  }
  &__duration {
    font-size: 1.1rem;
    position: absolute;
    bottom: 5%;
    right: 5%;
    background-color: rgba(250, 250, 250, 0.6);
    border-radius: 4px;
    backdrop-filter: blur(4px);
    padding: 2px 4px;
  }
  &__title {
    position: absolute;
    top: 5%;
    font-size: 1.1rem;
    width: 99.14%;
    background-color: rgba(250, 250, 250, 0.6);
    -webkit-backdrop-filter: blur(4px);
    backdrop-filter: blur(4px);
    padding: 2px 0px 2px 2px;
    box-shadow: rgba(0, 0, 0, 0.24) 0px 3px 8px;
  }
}

@include for-max-620px() {
  .workout-video {
    width: 65%;
    &__title {
      font-size: 1.3rem;
    }
    &__duration {
      font-size: 1.3rem;
    }
  }
}

@include for-max-440px() {
  .workout-video {
    width: 80%;
  }
}