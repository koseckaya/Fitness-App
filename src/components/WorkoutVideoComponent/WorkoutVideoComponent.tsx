import React from 'react'
import './WorkoutVideoComponent.scss';
import { FC } from 'react';
import { Video } from '../../data';

export const WorkoutVideoComponent: FC<Video> = ({title, src, srcImg, duration}) => {
        return (
            <div className='workout-video'>
                <a href={ src }>
                    <img className='workout-video__img' src={ srcImg } alt={ title } />
                    <div className='workout-video__title'>{title}</div>
                    <div className='workout-video__duration'>{ duration } MIN</div>
                </a>
            </div>
    )
}
   

