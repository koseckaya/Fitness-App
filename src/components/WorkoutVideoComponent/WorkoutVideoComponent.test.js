import React from 'react';
import renderer from 'react-test-renderer';
import { WorkoutVideoComponent } from './WorkoutVideoComponent';

const props = {
    title: '5 Min Warm Up',
    src: 'https://www.youtube.com/watch?v=7t1qhpDLSi0&ab_channel=ChloeTing',
    srcImg: 'https://static.chloeting.com/videos/63ae100649cf64962cb594e2/f081bd90-8a98-11ed-97ac-4b900c40c009.jpeg',
    duration: 5
}

describe("WorkoutVideoComponent component", () => {
    it('should render component with default props', () => {
        const tree = renderer
            .create(<WorkoutVideoComponent {...props} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render component without title props', () => {
        const newProps = { ...props, title: '' }
        const component = renderer
            .create(<WorkoutVideoComponent {...newProps} />)
            .toJSON();
        expect(component).toMatchSnapshot();
    });
    it('should render component without src props', () => {
        const newProps = { ...props, src: '' }
        const component = renderer
            .create(<WorkoutVideoComponent {...newProps} />)
            .toJSON();
        expect(component).toMatchSnapshot();
    });
    it('should render component without srcImg props', () => {
        const newProps = { ...props, srcImg: '' }
        const component = renderer
            .create(<WorkoutVideoComponent {...newProps} />)
            .toJSON();
        expect(component).toMatchSnapshot();
    });
    it('should render component without duration props', () => {
        const newProps = { ...props, duration: 0 }
        const component = renderer
            .create(<WorkoutVideoComponent {...newProps} />)
            .toJSON();
        expect(component).toMatchSnapshot();
    });
})