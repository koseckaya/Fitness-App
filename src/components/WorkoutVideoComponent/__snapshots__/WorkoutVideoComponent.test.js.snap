// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkoutVideoComponent component should render component with default props 1`] = `
<div
  className="workout-video"
>
  <a
    href="https://www.youtube.com/watch?v=7t1qhpDLSi0&ab_channel=ChloeTing"
  >
    <img
      alt="5 Min Warm Up"
      className="workout-video__img"
      src="https://static.chloeting.com/videos/63ae100649cf64962cb594e2/f081bd90-8a98-11ed-97ac-4b900c40c009.jpeg"
    />
    <div
      className="workout-video__title"
    >
      5 Min Warm Up
    </div>
    <div
      className="workout-video__duration"
    >
      5
       MIN
    </div>
  </a>
</div>
`;

exports[`WorkoutVideoComponent component should render component without duration props 1`] = `
<div
  className="workout-video"
>
  <a
    href="https://www.youtube.com/watch?v=7t1qhpDLSi0&ab_channel=ChloeTing"
  >
    <img
      alt="5 Min Warm Up"
      className="workout-video__img"
      src="https://static.chloeting.com/videos/63ae100649cf64962cb594e2/f081bd90-8a98-11ed-97ac-4b900c40c009.jpeg"
    />
    <div
      className="workout-video__title"
    >
      5 Min Warm Up
    </div>
    <div
      className="workout-video__duration"
    >
      0
       MIN
    </div>
  </a>
</div>
`;

exports[`WorkoutVideoComponent component should render component without src props 1`] = `
<div
  className="workout-video"
>
  <a
    href=""
  >
    <img
      alt="5 Min Warm Up"
      className="workout-video__img"
      src="https://static.chloeting.com/videos/63ae100649cf64962cb594e2/f081bd90-8a98-11ed-97ac-4b900c40c009.jpeg"
    />
    <div
      className="workout-video__title"
    >
      5 Min Warm Up
    </div>
    <div
      className="workout-video__duration"
    >
      5
       MIN
    </div>
  </a>
</div>
`;

exports[`WorkoutVideoComponent component should render component without srcImg props 1`] = `
<div
  className="workout-video"
>
  <a
    href="https://www.youtube.com/watch?v=7t1qhpDLSi0&ab_channel=ChloeTing"
  >
    <img
      alt="5 Min Warm Up"
      className="workout-video__img"
      src=""
    />
    <div
      className="workout-video__title"
    >
      5 Min Warm Up
    </div>
    <div
      className="workout-video__duration"
    >
      5
       MIN
    </div>
  </a>
</div>
`;

exports[`WorkoutVideoComponent component should render component without title props 1`] = `
<div
  className="workout-video"
>
  <a
    href="https://www.youtube.com/watch?v=7t1qhpDLSi0&ab_channel=ChloeTing"
  >
    <img
      alt=""
      className="workout-video__img"
      src="https://static.chloeting.com/videos/63ae100649cf64962cb594e2/f081bd90-8a98-11ed-97ac-4b900c40c009.jpeg"
    />
    <div
      className="workout-video__title"
    />
    <div
      className="workout-video__duration"
    >
      5
       MIN
    </div>
  </a>
</div>
`;
